packages/core/src/shared/lsp/lspResolver.ts

```diff
@@ -47,25 +47,40 @@ export class LanguageServerResolver {
 
             const serverResolvers: StageResolver<LspResult>[] = [
                 {
+                    // 1: Use the current local ("cached") LSP server bundle, if any.
                     resolve: async () => await this.getLocalServer(cacheDirectory, latestVersion, targetContents),
                     telemetryMetadata: { id: this.lsName, languageServerLocation: 'cache' },
                 },
                 {
+                    // 2: Download the latest LSP server bundle.
                     resolve: async () => await this.fetchRemoteServer(cacheDirectory, latestVersion, targetContents),
```

Comment:
It seems like these 3 functions (`getFallbackServer`, `fetchRemoteServer`, `getLocalServer`) are all variations on the same thing, but their names don't make that obvious.

Reply:
Do you see a better naming convention here? My intention here was to make it clear where the server was coming from by following a  `[verb][source]Server` pattern where `verb` follows from source.

Reply:
Thats a good point. I was trying to have the verb be consistent with the operation being done, but perhaps this can be abstracted from the caller. 

This `Stage` abstraction was mostly for telemetry and to be clearly consistent with the LSP Spec.